packages/grid/x-data-grid/src/hooks/utils/useGridApiMethod.ts

```diff
@@ -13,31 +13,22 @@ export function useGridApiMethod<
   T extends V extends 'public' ? Partial<PublicApi> : Partial<PrivateOnlyApi>,
 >(privateApiRef: React.MutableRefObject<PrivateApi>, apiMethods: T, visibility: V) {
   const apiMethodsRef = React.useRef(apiMethods);
-  const [apiMethodsNames] = React.useState(Object.keys(apiMethods) as Array<keyof T>);
 
+  /* eslint-disable react-hooks/exhaustive-deps */
   const installMethods = React.useCallback(() => {
-    if (!privateApiRef.current) {
-      return;
-    }
-    apiMethodsNames.forEach((methodName) => {
-      if (!privateApiRef.current.hasOwnProperty(methodName)) {
-        privateApiRef.current.register(visibility, {
-          [methodName]: (...args: any[]) => {
-            const fn = apiMethodsRef.current[methodName] as Function;
-            return fn(...args);
-          },
-        });
-      }
+    Object.keys(apiMethodsRef.current).forEach((methodName) => {
+      privateApiRef.current.register(visibility, {
+        [methodName]: apiMethodsRef.current[methodName as keyof typeof apiMethods],
+      });
     });
-  }, [apiMethodsNames, privateApiRef, visibility]);
+  }, [privateApiRef, visibility, apiMethods]);
+  /* eslint-enable react-hooks/exhaustive-deps */
```

Comment:
Is the goal here to register methods every time the `apiMethods` reference changes?
If so, you can get rid of `apiMethodsRef` and use `apiMethods` directly

Reply:
> Is the goal here to register methods every time the apiMethods reference changes?

Yes.

> Is the goal here to register methods every time the apiMethods reference changes?
If so, you can get rid of apiMethodsRef and use apiMethods directly

I tried. It didn't go well. I spent a few hours trying to use `apiMethods` directly but it was creating issues in seemingly unrelated parts of the grid. I could try again.

Reply:
Yeah, I see what you mean. Somehow it only works, when we register values in the effect, not during render.
I managed to simplify it to this:
```tsx
  const isFirstRender = React.useRef(true);
  React.useEffect(() => {
    isFirstRender.current = false;
    privateApiRef.current.register(visibility, apiMethods);
  }, [privateApiRef, visibility, apiMethods]);

  if (isFirstRender.current) {
    privateApiRef.current.register(visibility, apiMethods);
  }
```

Reply:
Thanks a lot, this probably saved hours of my life. I was going crazy with that piece of code.